script.js

```diff
@@ -1,4 +1,200 @@
+  let randomNum = Math.floor(Math.random() * 6) + 1;
+  return randomNum;
+}
+// arrange the order of dices as per user input.
+function arrangeDice(userInput, firstDice, secondDice) {
+  let arrangedResult;
+  if (userInput == 'dice1') {
+    diceOrder = 'dice1';
+  }
+  else {
+    diceOrder = 'dice2';
+  }
+  if (diceOrder == 'dice1') {
+    arrangedResult = `${firstDice}${secondDice}`;
+  }
+  else {
+    arrangedResult = `${secondDice}${firstDice}`;
+  }
+  return Number(arrangedResult);
+}
+// function is used for finding winner with highest (normal) as well as lowest score modes.
+function findWinner(playersScoreArray, mode) {
+  if (mode == 'normal') {
+    return Math.max.apply(null, playersScoreArray);
+  }
+  return Math.min.apply(null, playersScoreArray);
+}
+function autoGenNumber(rolledNumArray) {
+  let array = rolledNumArray;
+  let finalNumber = 0;
+  let length = array.length;
+  for (let i = 0; array.length >= 1; i++) {
+    let max = Math.max.apply(null, array);
+    finalNumber = finalNumber * 10 + max;
+    rolledNumArray = array.splice(array.indexOf(max), 1);
+  }
+  return finalNumber;
+}
+function setMode(userInput) {
+  let mode;
+  if (userInput == 'manual') {
+    mode = 'manual';
+  }
+  else if (userInput == 'auto') {
+    mode = 'auto';
+  }
+  return mode;
+}
+function iterateNumOfPlayers(totalPlayers, mode, userInput) {
+  let currentPlayer; let nextPlayer;
+  let playersScoreArray = [];
+  let playerDices = [];
+  let resultedNumber;
+  for (let i = 1; i <= totalPlayers; i++) {
+    let dice1 = rollDice();
+    let dice2 = rollDice();
+    currentPlayer = 'player' + [i];
+    if (i + 1 <= totalPlayers) {
+      nextPlayer = 'player' + [i + 1];
+    }
+    if (mode == 'auto') {
+      playerDices.push(dice1);
+      playerDices.push(dice2);
+      resultedNumber = autoGenNumber(playerDices);
+    }
+    else if (mode == 'manual') {
+      console.log('choosing mode works');
+      resultedNumber = arrangeDice(UserInput, dice1, dice2);
+    }
+    playersScoreArray.push(resultedNumber);
+    playerDices = [];
+  }
+  console.log('players score array: ' + playersScoreArray);
+  return playersScoreArray;
+}
+function rollDiceNTimes(numberOfDices) {
+  let diceArray = [];
+  for (let i = 0; i < numberOfDices; i++) {
+    let num = rollDice();
+    diceArray.push(num);
+  }
+  return diceArray;
+}
+function generateNumFromIndex(userInput, rolledArray) {
+  let finalNumber = 0;
+  let indexNumber = userInput.split('').map(Number);
+  console.log('indexNumber: ' + indexNumber);
+  for (let i = 0; i < indexNumber.length; i++) {
+    finalNumber = finalNumber * 10 + rolledArray[indexNumber[i]];
+  }
+  return finalNumber;
+}
+let firstRoll; let secondRoll;
+let currentTurn = 'player1'; let changedTurn;
+let diceOrder;
+let roundsPlayed = 0;
+let mode = 'set num of players';
+let numOfPlayers = 0;
+let playerDices = [];
+let autoArrange = [];
+let numOfDices = 0;
+let playersArray;
+let nDiceMode = false;
```

Comment:
1 var declaration per line. Please add comments to explain what each of these are used for, and try to group them for easier reading.